### PR TITLE
add in axios for http requests

### PR DESCRIPTION
**Pull request by cadebward**
tldr; use axios instead of native fetch for http requests

1) native fetch handles errors poorly

For example, if you were to hit a page that does not exist, you would expect that fetch would throw. This is not the case. You would actually get back a resolved promise with a status of 400.

2) native fetch requires the user to manually convert response into something usable

3) axios is HUGELY OMEGA MASSIVELY used by the community. Its not very large, and it provides an incredible developer experience. Its fully tested, fully documented, and does everything FS.fetch is doing for us, plus more.

p.s. i don't know much about our auth yet, but yes, axios provides an easy way to attach headers to all requests so we don't have to manually authenticate each http call. I plan to learn more about this and provide something in lib that does that out of the box.

### CODE REVIEW

**Comment by redbugz** [comment]
We already have started on a fetch wrapper that does the improved error handling and auth and things like that. We've also used superagent over axios in the past. Though we expect to deprecate superagent once we have a good solid fetch wrapper.
I'm not opposed to axios, but it would be interesting to compare and see what benefits it provides. I know when I used it in the past I was frustrated with it's lack of debug and logging options, but that was a long time ago.

**Comment by joeycozza** [comment]
I think part of the appeal is not having to write/maintain a fetch wrapper.
but axios doesn't handle service workers, I wonder if that is a big deal for us.

**Comment by joeycozza** [comment]
I think we let people npm install axios as needed, and then they also can use @fs/axios-defaults in conjunction with it.
